mm2src/mm2_main/src/lp_swap/swap_v2_common.rs

```diff
@@ -371,3 +372,65 @@ pub(super) async fn swap_kickstart_handler<
         error!("Error {} on trying to run the swap {}", e, uuid);
     }
 }
+
+pub(super) async fn swap_kickstart_handler_for_maker(
+    ctx: MmArc,
+    swap_repr: MakerSwapDbRepr,
+    storage: MakerSwapStorage,
+    uuid: Uuid,
+) {
+    if let Some((maker_coin, taker_coin)) = swap_kickstart_coins(&ctx, &swap_repr, &uuid).await {
+        match (maker_coin, taker_coin) {
+            (MmCoinEnum::UtxoCoin(m), MmCoinEnum::UtxoCoin(t)) => {
+                swap_kickstart_handler::<MakerSwapStateMachine<_, _>, _, _>(swap_repr, storage, uuid, m, t).await
+            },
+            (MmCoinEnum::EthCoin(m), MmCoinEnum::EthCoin(t)) => {
+                swap_kickstart_handler::<MakerSwapStateMachine<_, _>, _, _>(swap_repr, storage, uuid, m, t).await
+            },
+            (MmCoinEnum::UtxoCoin(m), MmCoinEnum::EthCoin(t)) => {
+                swap_kickstart_handler::<MakerSwapStateMachine<_, _>, _, _>(swap_repr, storage, uuid, m, t).await
+            },
+            (MmCoinEnum::EthCoin(m), MmCoinEnum::UtxoCoin(t)) => {
+                swap_kickstart_handler::<MakerSwapStateMachine<_, _>, _, _>(swap_repr, storage, uuid, m, t).await
+            },
+            _ => {
+                error!(
+                    "V2 swaps are not currently supported for {}/{} pair",
+                    swap_repr.maker_coin(),
+                    swap_repr.taker_coin()
+                );
+            },
+        }
+    }
+}
+
+pub(super) async fn swap_kickstart_handler_for_taker(
```

Comment:
I am seeing code duplication in swap_kickstart_handler_for_taker and swap_kickstart_handler_for_maker.
I tried to fix it but it looks like this won't happen w/o considerable refactoring with using trait objects, right?

Reply:
> I am seeing code duplication in swap_kickstart_handler_for_taker and swap_kickstart_handler_for_maker. I tried to fix it but it looks like this won't happen w/o considerable refactoring with using trait objects, right?

Yep, it is related to the issue where MmCoinEnum matching for coins pair requires explicit types to be set in match arms. Otherwise, it leads to compiler errors.